lib/io/fileutils/fileutils.go

```diff
@@ -175,3 +187,98 @@ func CreateTempFile(content string, filename string) (string, error) {
 func DeleteTempFile(tempFile string) error {
 	return os.Remove(tempFile)
 }
+
+// Moves file from current/source directory to destination directory.
+// Creates the directory, if not already exists
+//
+// Example usage:
+// err := fileutils.Move("file.txt", "/path/to/my/dir/", "");
+// Incase of retriving the old file name, leave renamedFileName as empty
+func Move(sourceFile string, destinationFile string) error {
+	destinationDir, fileName := filepath.Split(destinationFile)
+
+	if err := os.MkdirAll(destinationDir, os.ModePerm); err != nil {
+		return err
+	}
+	err := os.Rename(sourceFile, destinationFile)
+	if err != nil {
+		// Checks if the error is of type *os.LinkError
+		if linkErr, ok := err.(*os.LinkError); ok {
+			// Checks the underlying error code & Handles the "file exists" error
+			if linkErr.Err.Error() == "file exists" {
+				if err = os.Remove(destinationFile); err != nil {
+					return err
+				}
+				// Retrys the rename operation
+				if err = os.Rename(sourceFile, destinationDir+fileName); err != nil {
+					return err
+				}
+			}
+		}
+		return err
+	}
+	return nil
+}
+
+func RemoveFirstLine(filePath string) error {
+	// Open the file for reading
+	file, err := os.OpenFile(filePath, os.O_RDWR, 0644)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	// Create a temporary file to write the updated content
+	tempFile, err := os.CreateTemp("", "temp")
+	if err != nil {
+		return err
+	}
+	defer tempFile.Close()
+
+	// Create a scanner to read the file line by line
+	scanner := bufio.NewScanner(file)
+
+	// Skip the first line
+	scanner.Scan()
+
+	// Write the remaining lines to the temporary file
+	for scanner.Scan() {
+		line := scanner.Text()
+		_, err := tempFile.WriteString(line + "\n")
+		if err != nil {
+			return err
+		}
+	}
+
+	// Close the files
+	file.Close()
+	tempFile.Close()
+
+	// Replace the original file with the temporary file
+	if err = os.Rename(tempFile.Name(), filePath); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// createTomlFileFromConfig created a toml file where path and struct interface is provided
+func CreateTomlFileFromConfig(config interface{}, tomlFile string) (string, error) {
+	f, err := os.Create(tomlFile)
+
+	if err != nil {
+		// failed to create/open the file
+		errors.Wrap(err, "Failed to create/open the file, \n%v")
+		return "", err
+	}
+	if err := toml.NewEncoder(f).Encode(config); err != nil {
+		// failed to encode
+		errors.Wrap(err, "Failed to encode\n%v")
+		return "", err
```

Comment:
can return at the same line, errors.wrap will return error

Reply:
Done